cmd/derper/derper.go

```diff
@@ -36,6 +36,7 @@ import (
 var (
 	dev           = flag.Bool("dev", false, "run in localhost development mode")
 	addr          = flag.String("a", ":443", "server address")
+	httpPort      = flag.String("http-port", "80", "The port on which to serve HTTP. Set to 0 to disable")
```

Comment:
0 for ports typically means auto-select. Since you used a String, how about saying the empty string means to not run it instead.

Reply:
On this aspect, I followed Kubernetes guidance which seems to use `0` as a disable value for this kind of listening handler notably on the `kube-apiserver` with CLI flags like `--insecure=port=0` or on the `kubelet` with `--healthz-port int32     Default: 10248`. They all use `set to 0 to disable`.
I've never seen `0` as auto-select behavior, do you have an example in mind ?

Reply:
also `--http-port=""` or `--http-port` cli flags looks a bit confusing to me to disable said port

Reply:
With all operating systems, listening on port 0 asks the operating system to pick a random port.

See https://www.google.com/search?q=port+0+wildcard+listen+random+port with various links.

We use & depend on that for e.g. integration tests: https://github.com/tailscale/tailscale/blob/v1.18.1/tstest/integration/integration_test.go#L710

Then our test runner just observes what port number was given to us by the kernel.

Kubernetes is apparently special-casing 0 to mean "disabled". I hadn't seen that style before.

Since it's still a string, how about special-casing the word `"off"`? Does that read nicer? `--http-port=off`?

Reply:
Well, I was not aware of `0` meaning asking the kernel for a random free port. Thanks for pointing that out
Then, if you rely on this behavior I guess the best value is to use `-1` as you suggested from the start which is a widespread "disable this" value.
I'll also set the flag back as an `Int`